src/analysis/normed_space/hahn_banach.lean

```diff
@@ -95,48 +124,53 @@ begin
   { exact f.op_norm_le_bound g.extend_to_ℂ.op_norm_nonneg (λ x, h x ▸ g.extend_to_ℂ.le_op_norm x) },
 end
 
+instance complex_has_exists_extension_norm_eq : has_exists_extension_norm_eq ℂ :=
+⟨by { intros, apply complex.exists_extension_norm_eq }⟩
+
 end complex
 
-section dual_vector
-variables {E : Type*} [normed_group E] [normed_space ℝ E]
+variables {𝕜 : Type u} [nondiscrete_normed_field 𝕜] [normed_algebra ℝ 𝕜]
+variables {E : Type v} [normed_group E] [normed_space 𝕜 E]
 
 open continuous_linear_equiv
 open_locale classical
 
-lemma coord_self' (x : E) (h : x ≠ 0) : (∥x∥ • (coord ℝ x h))
-  ⟨x, submodule.mem_span_singleton_self x⟩ = ∥x∥ :=
-calc (∥x∥ • (coord ℝ x h)) ⟨x, submodule.mem_span_singleton_self x⟩
-    = ∥x∥ • (linear_equiv.coord ℝ E x h) ⟨x, submodule.mem_span_singleton_self x⟩ : rfl
-... = ∥x∥ • 1 : by rw linear_equiv.coord_self ℝ E x h
-... = ∥x∥ : mul_one _
+lemma coord_self' (x : E) (h : x ≠ 0) : (coe_from_ℝ 𝕜 (∥x∥) • (coord 𝕜 x h))
+  ⟨x, submodule.mem_span_singleton_self x⟩ = coe_from_ℝ 𝕜 (∥x∥) :=
+calc ((coe_from_ℝ 𝕜 (∥x∥)) • (coord 𝕜 x h)) ⟨x, submodule.mem_span_singleton_self x⟩
+    = (coe_from_ℝ 𝕜 (∥x∥)) • (linear_equiv.coord 𝕜 E x h) ⟨x, submodule.mem_span_singleton_self x⟩ : rfl
+... = (coe_from_ℝ 𝕜 (∥x∥)) • 1 : by rw linear_equiv.coord_self 𝕜 E x h
+... = (coe_from_ℝ 𝕜 (∥x∥)) : mul_one _
+
+lemma coord_norm' (x : E) (h : x ≠ 0) : ∥(coe_from_ℝ 𝕜 (∥x∥)) • coord 𝕜 x h∥ = 1 :=
+by rw [norm_smul, norm_norm', coord_norm, mul_inv_cancel (mt norm_eq_zero.mp h)]
 
-lemma coord_norm' (x : E) (h : x ≠ 0) : ∥∥x∥ • coord ℝ x h∥ = 1 :=
-by rw [norm_smul, norm_norm, coord_norm, mul_inv_cancel (mt norm_eq_zero.mp h)]
+variables [has_exists_extension_norm_eq.{u v} 𝕜]
```

Comment:
Needed to avoid errors like
```
failed to synthesize type class instance for
𝕜 : Type u,
_inst_3 : has_exists_extension_norm_eq.{u u_1} 𝕜,
⊢ has_exists_extension_norm_eq.{u v} 𝕜
```

Not sure if there's a better way.

Reply:
As long as `E` is hidden behind a `∀` and has a different universe parameter than `𝕜`, I think we cannot avoid writing out the universes explicitly. One trick is to swap the order of `u` and `v` in the `universes u v` declaration, then we only need to give `v` explicitly: `[has_exists_extension_norm_eq.{v}]`.